src/internal/connector/onedrive/collections.go

```diff
@@ -434,15 +466,62 @@ func (c *Collections) UpdateCollections(
 			// already created and partially populated.
 			updatePath(newPaths, *item.GetId(), folderPath.String())
 
+			if ok {
+				// If we find prev path, that implies these are
+				// moved/renames folders and not new ones
+				prevPath, err := path.FromDataLayerPath(prevColPath, true) // FIXME(meain): converting string to path
+				if err != nil {
+					logger.Ctx(ctx).Errorw("invalid prev path for item", "error", err)
+					return err
+				}
+
+				col, found := c.CollectionMap[folderPath.String()]
+				if !found {
+					col := NewCollection(
+						c.itemClient,
+						folderPath,
+						prevPath,
+						driveID,
+						c.service,
+						c.statusUpdater,
+						c.source,
+						c.ctrl,
+						invalidPrevDelta,
+					)
+					c.CollectionMap[folderPath.String()] = col
+					c.NumContainers++
+				} else {
+					// If we have previously created a collection when
+					// we encountered a file, this gives us a change
+					// to mark the container as moved or not moved
+					// instead of new.
+					ccol, ok := col.(*Collection)
+					if !ok {
+						logger.Ctx(ctx).Errorw("unable to cast onedrive collection", "error", err)
+						return err
+					}
+
+					ccol.SetPreviousPath(prevPath)
+					c.CollectionMap[folderPath.String()] = ccol
```

Comment:
this could result in the collection ending up in the map multiple times maybe (see earlier comment about maybe picking up both "halves" of a folder move). We should probably do the following in these cases:
1. get the current path of the collection
2. set the collection's FullPath path to `folderPath`
3. `delete(c.CollectionMap, previousFullPath.String())`
4. add the folder to the map at the current path

Reply:
I think the same [comment about delta only returning latest state](https://github.com/alcionai/corso/pull/2407/files#r1098458571) applies here as well.